ViewControllers/DetailViewController.swift

```diff
@@ -202,40 +238,67 @@ class DetailViewController: UIViewController {
+        contentView.addSubview(titleLabel)
+        contentView.addSubview(imageView)
+        contentView.addSubview(releaseStack)
+        contentView.addSubview(durationStack)
+        contentView.addSubview(movieDescriptionLabel)
+        contentView.addSubview(collectionView)
         
         NSLayoutConstraint.activate([
-            titleLabel.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor),
-            titleLabel.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor),
-            titleLabel.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor),
+            titleLabel.topAnchor.constraint(equalTo: contentView.layoutMarginsGuide.topAnchor),
+            titleLabel.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor),
+            titleLabel.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor),
 
             imageView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 16),
             imageView.centerXAnchor.constraint(equalTo: titleLabel.centerXAnchor),
-            imageView.widthAnchor.constraint(equalTo: view.widthAnchor, multiplier: 0.61),
+            imageView.widthAnchor.constraint(equalTo: contentView.widthAnchor, multiplier: 0.61),
             imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: 1.41),
             
-            releaseStack.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor),
+            releaseStack.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor),
             releaseStack.topAnchor.constraint(equalTo: imageView.bottomAnchor, constant: 35),
 
             durationStack.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             durationStack.topAnchor.constraint(equalTo: releaseStack.bottomAnchor, constant: 8),
 
             movieDescriptionLabel.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
-            movieDescriptionLabel.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor),
+            movieDescriptionLabel.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor),
             movieDescriptionLabel.topAnchor.constraint(equalTo: durationStack.bottomAnchor, constant: 16),
 
             collectionView.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             collectionView.topAnchor.constraint(equalTo: movieDescriptionLabel.bottomAnchor, constant: 16),
             collectionView.rightAnchor.constraint(equalTo: movieDescriptionLabel.rightAnchor),
-            collectionView.heightAnchor.constraint(equalToConstant: 200),
+            collectionView.heightAnchor.constraint(equalToConstant: calculateCollectionViewHeight())
         ])
     }
     
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        scrollView.contentSize = CGSize(width: view.frame.size.width, height: calculateScrollViewContentSize())
```

Comment:
We are only needing to do this here because our `contentView` doesn't know its size. We are missing the link between the bottom of the last view and the bottom of our content view. After that is set, this work will be automatically done by the layout engine itself